File: Python/0050. Theatre Square.py
# If you need to import additional packages or classes, please import here.

def func():
    # please define the python3 input here. 
    # For example: a,b = map(int, input().strip().split())
    # please finish the function body here.
    # please define the python3 output here. For example: print().
    while True:
        try:
            m,n,a = map(int,input().strip().split())
            print((m//a+min(1,m%a))*(n//a+min(1,n%a)))
            
        except EOFError:
            break
    
if __name__ == "__main__":
    func()

File: C++/0050. Theatre Square.cpp
// we have defined the necessary header files here for this problem.
// If additional header files are needed in your program, please import here.

int main()
{  
  // please define the C++ input here. For example: int a,b; cin>>a>>b;;  
  // please finish the function body here.  
  // please define the C++ output here. For example:cout<<____<<endl; 

    long long m,n,a;
    while(cin>>m>>n>>a){
        cout<<(m/a+min(1LL,m%a))*(n/a+min(1LL,n%a))<<endl;
    }
    return 0;
}
